iotilecore/iotile/core/utilities/async_tools/event_loop.py

```diff
@@ -692,6 +735,25 @@ def _log_future_exception(future, logger):
         logger.warning("Exception in ignored future: %s", future, exc_info=True)
 
 
+async def _repeat(cor, interval, loop, *args, **kwargs):
+    """Run a coroutine every interval seconds.
+
+    Note that if your function doesn't finish in time, it will run again immediately
+    after the previous iteration is finished (it doesn't guarantee runtime).
+
+    Each repeatable will have its own tick generation to account for its own drift.
+    """
+    tg = _tick_generator(interval, loop)
+    while 1:
+        await asyncio.gather(cor(*args, **kwargs), asyncio.sleep(next(tg)),)
```

Comment:
If `cor` raises an exception, the repeating loop will be stopped and no more invocations will happen.  Is that the desired behavior?  I could see it both ways but likely I think the expectation would be that the repeat loop keeps going even if one particular invocation raised an exception.

Reply:
For observability, I would recommend logging the exception as well.

Reply:
Good point, yes. I amended the function signature to accept the argument `handle_exceptions`, defaulting to True. The behavior now is that unexpected exceptions will be logged, and unless the exception was an `asyncio.CancelledError` (which would be how a user manually cancels their loop). If you have a case where you want the exception to break out of the loop (for example, during application development), you'd just set the handler to false.